web-backend/src/main/java/com/statusneo/vms/service/OtpService.java

```diff
@@ -61,13 +82,47 @@ public Optional<Otp> getLatestOtpByEmail(String email) {
         return otps.isEmpty() ? Optional.empty() : Optional.of(otps.getFirst());
     }
 
+    // Track OTP attempts per email
+
+    /**
+     * Resends a new OTP to the given email if at least 2 minutes have passed since
+     * the last OTP was sent.
+     *
+     * @param email The email to resend the OTP to.
+     * @return true if OTP was resent, false if not enough time has passed.
+     */
+    public boolean canResendOtp(String email) {
+        Optional<Otp> latestOtp = getLatestOtpByEmail(email);
```

Comment:
I feel like there's a problem here. You're tying OTP to visitor email which is logically correct but technically OTP is tied to a visit entity. It should be as simple as a visit lookup, while latestOTPByEmail will do a column or index scan.